src/inmanta/reporter.py

```diff
@@ -0,0 +1,149 @@
+# Adapted from pyformance
+
+import asyncio
+import time
+import base64
+import logging
+
+from pyformance import global_registry
+from tornado.httpclient import AsyncHTTPClient, HTTPRequest, HTTPError
+from urllib.parse import quote
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_INFLUX_SERVER = "127.0.0.1"
+DEFAULT_INFLUX_PORT = 8086
+DEFAULT_INFLUX_DATABASE = "metrics"
+DEFAULT_INFLUX_USERNAME = None
+DEFAULT_INFLUX_PASSWORD = None
+DEFAULT_INFLUX_PROTOCOL = "http"
+
+
+class AsyncReporter(object):
+    def __init__(self, registry=None, reporting_interval=30, clock=None):
+        self.registry = registry or global_registry()
+        self.reporting_interval = reporting_interval
+        self.clock = clock or time
+        self._stopped = False
+        self._handle = False
+
+    def start(self):
+        if self._stopped:
+            return False
+        self._handle = asyncio.get_event_loop().create_task(self._loop())
+        return True
+
+    def stop(self):
+        self._stopped = True
+        self._handle.cancel()
+
+    async def _loop(self):
+        loop = asyncio.get_event_loop()
+        next_loop_time = loop.time()
+        while not self._stopped:
+            try:
+                await self.report_now(self.registry)
+            except Exception:
+                LOGGER.warning("Could not send metrics report", exc_info=True)
+            next_loop_time += self.reporting_interval
+            wait = max(0, next_loop_time - time.time())
+            await asyncio.sleep(wait)
+
+    async def report_now(self, registry=None, timestamp=None):
+        raise NotImplementedError(self.report_now)
+
+
+class InfluxReporter(AsyncReporter):
+
+    """
+    InfluxDB reporter using native http api
+    (based on https://influxdb.com/docs/v1.1/guides/writing_data.html)
+    """
+
+    def __init__(
+        self,
+        registry=None,
+        reporting_interval=5,
+        prefix="",
+        database=DEFAULT_INFLUX_DATABASE,
+        server=DEFAULT_INFLUX_SERVER,
+        username=DEFAULT_INFLUX_USERNAME,
+        password=DEFAULT_INFLUX_PASSWORD,
+        port=DEFAULT_INFLUX_PORT,
+        protocol=DEFAULT_INFLUX_PROTOCOL,
+        autocreate_database=False,
+        clock=None,
+        tags={},
+    ):
+        super(InfluxReporter, self).__init__(registry, reporting_interval, clock)
+        self.prefix = prefix
+        self.database = database
+        self.username = username
+        self.password = password
+        self.port = port
+        self.protocol = protocol
+        self.server = server
+        self.autocreate_database = autocreate_database
+        self._did_create_database = False
+        self.tags = {}
+        self.key = "metrics"
```

Comment:
This should be "tags" instead of "self.tags". The same for the line below "self.tags" should be "tags.items()"

Reply:
thank you! missed that completely